modules/hostap/src/supp_main.c

```diff
@@ -0,0 +1,546 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+#include <sys/fcntl.h>
+LOG_MODULE_REGISTER(wpa_supplicant, LOG_LEVEL_DBG);
+
+#if defined(CONFIG_WPA_SUPP_CRYPTO) && !defined(CONFIG_MBEDTLS_ENABLE_HEAP)
+#include <mbedtls/platform.h>
+#endif /* CONFIG_WPA_SUPP_CRYPTO */
+
+#include <zephyr/net/socket.h>
+#include <zephyr/net/socket_select.h>
+#include <zephyr/net/wifi_nm.h>
+
+#include "includes.h"
+#include "common.h"
+#include "eloop.h"
+#include "wpa_supplicant/config.h"
+#include "wpa_supplicant_i.h"
+
+#include "fst/fst.h"
+#include "includes.h"
+#include "p2p_supplicant.h"
+#include "driver_i.h"
+
+#include "supp_main.h"
+#include "supp_events.h"
+#include "wpa_cli_zephyr.h"
+
+#include "supp_api.h"
+
+K_SEM_DEFINE(z_wpas_ready_sem, 0, 1);
+#include <l2_packet/l2_packet.h>
+
+/* Should match with the driver name */
+#define DEFAULT_IFACE_NAME "wlan0"
+#define IFACE_MATCHING_PREFIX "wlan"
+
+static struct net_mgmt_event_callback cb;
+struct k_mutex iface_up_mutex;
+
+struct wpa_global *global;
+
+static int z_wpas_event_sockpair[2];
+
+static void z_wpas_start(void);
+static void z_wpas_iface_work_handler(struct k_work *item);
+
+static K_THREAD_STACK_DEFINE(z_wpa_s_thread_stack,
+			     CONFIG_WIFI_NM_WPA_SUPPLICANT_THREAD_STACK_SIZE);
+static struct k_thread z_wpa_s_tid;
+
+static K_THREAD_STACK_DEFINE(z_wpas_iface_wq_stack,
+	CONFIG_WIFI_NM_WPA_SUPPLICANT_WQ_STACK_SIZE);
+
+/* TODO: Debug why wsing system workqueue blocks the driver dedicated
+ * workqueue?
+ */
+static struct k_work_q z_wpas_iface_wq;
+static K_WORK_DEFINE(z_wpas_iface_work,
+	z_wpas_iface_work_handler);
+
+K_MUTEX_DEFINE(z_wpas_event_mutex);
+
+static const struct wifi_mgmt_ops wpa_supp_ops = {
+	.scan = z_wpa_supplicant_scan,
+	.connect = z_wpa_supplicant_connect,
+	.disconnect = z_wpa_supplicant_disconnect,
+	.iface_status = z_wpa_supplicant_status,
+#ifdef CONFIG_NET_STATISTICS_WIFI
+	.get_stats = z_wpa_supplicant_get_stats,
+#endif
+	.set_power_save = z_wpa_supplicant_set_power_save,
+	.set_twt = z_wpa_supplicant_set_twt,
+	.get_power_save_config = z_wpa_supplicant_get_power_save_config,
+	.reg_domain = z_wpa_supplicant_reg_domain,
+};
+
+DEFINE_WIFI_NM_INSTANCE(wpa_supplicant, &wpa_supp_ops);
+
+#ifdef CONFIG_MATCH_IFACE
```

Comment:
This isn't used and not really applicable for Zephyr, so, can be removed.

Reply:
Hi @krish2718, just asking that did you review some older version because I do not have that code in my tree, it was removed already in the latest update.

Reply:
Ah you are right, its gone now :).